Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -74,4 +74,47 @@ lemma changeLevel_trans {m d : ℕ} (hm : n ∣ m) (hd : m ∣ d) :
     changeLevel (dvd_trans hm hd) χ = changeLevel hd (changeLevel hm χ) := by
   simp [changeLevel_def, MonoidHom.comp_assoc, ZMod.unitsMap_comp]
 
+lemma unitsMap_comp {d m : ℕ} (hm : n ∣ m) (hd : m ∣ d) :
+    (ZMod.unitsMap hm).comp (ZMod.unitsMap hd) = ZMod.unitsMap (dvd_trans hm hd) := by
+  simp only [ZMod.unitsMap_def]
+  rw [← Units.map_comp]
+  exact congr_arg Units.map <| congr_arg RingHom.toMonoidHom <| ZMod.castHom_comp hm hd
+
+lemma ofUnitHom_eq {m : ℕ} (hm : n ∣ m) (a : Units (ZMod m)) :
+    (changeLevel hm χ) a = χ a := by
```

Comment:
The name of this lemma should refer to `changeLevel` instead of `ofUnitHom`.

Reply:
now I propose `changeLevel_eq_cast_of_dvd`. Might it also make change to put it into a `ChangeLevel` namespace?

Reply:
Not really; there is no argument whose type is `changeLevel ...`.